galley/pkg/source/fs/source.go

```diff
@@ -132,35 +133,42 @@ func (s *source) readFile(path string, info os.FileInfo, initial bool) map[resou
 }
 
 //process delete part of the resources in a file
-func (s *source) processPartialDelete(fileName string, newData *map[resource.FullName]*fileResource) {
+func (s *source) processPartialDelete(fileName string, newData []*fileResource) {
```

Comment:
Forgot to ask in the previous review: Is there a reason for changing from map to array? It looks like you'll need to loop through anyway.

Reply:
Yes. If its a map then 'resource.FullName' is not a sufficient key to avoid collisions across types. 

Also using an array gives predictable update ordering and is not subject to map randomization.